rxkeystore/src/main/java/com/nexenio/rxkeystore/provider/asymmetric/BaseAsymmetricCryptoProvider.java

```diff
@@ -106,15 +109,33 @@ public Completable verify(@NonNull byte[] data, @NonNull byte[] signature, @NonN
 
     @Override
     public Single<KeyPair> generateKeyPair(@NonNull String alias, @NonNull Context context) {
-        return getKeyPairGeneratorInstance()
-                .flatMap(keyPairGenerator -> getKeyAlgorithmParameterSpec(alias, context)
-                        .map(algorithmParameterSpec -> {
-                            keyPairGenerator.initialize(algorithmParameterSpec);
-                            return keyPairGenerator.generateKeyPair();
-                        }))
-                .onErrorResumeNext(throwable -> Single.error(
-                        new KeyGenerationException("Unable to generate key pair", throwable)
-                ));
+        return Single.just(rxKeyStore)
+                .flatMap(keyStore -> {
```

Comment:
Why is `rxKeyStore` part of the stream? Just use `Single.defer` and directly access `rxKeyStore`.